notifications/core/build.gradle

```
@@ -124,8 +124,8 @@ dependencies {
     implementation "commons-logging:commons-logging:${versions.commonslogging}"
     implementation "org.jetbrains.kotlin:kotlin-stdlib:${kotlin_version}"
     implementation "org.jetbrains.kotlin:kotlin-stdlib-common:${kotlin_version}"
-    implementation "org.apache.httpcomponents:httpcore:${versions.httpcore}"
-    implementation "org.apache.httpcomponents:httpclient:${versions.httpclient}"
+    implementation "org.apache.httpcomponents.core5:httpcore5:${versions.httpcore5}"
+    implementation "org.apache.httpcomponents.client5:httpclient5:${versions.httpclient5}"
     implementation "com.fasterxml.jackson.core:jackson-databind:${versions.jackson_databind}"
```

```suggestion
    compileOnly "com.fasterxml.jackson.core:jackson-databind:${versions.jackson_databind}"
    compileOnly "com.fasterxml.jackson.core:jackson-annotations:${versions.jackson}"
```

Otherwise, JarHell due to https://github.com/opensearch-project/OpenSearch/pull/5366

---

Is the OpenSearch 3.0 snapshot that's already available contain the change in the PR above already? I ran the build and test for the Notifications test in this PR already and didn't hit any jarhell issues.

---

Yes, it does

---

> I ran the build and test for the Notifications test in this PR already and didn't hit any jarhell issues.

Did you build the Docker container for `opensearch-notifications:test`?

---

[Our workflow](https://github.com/opensearch-project/notifications/blob/main/.github/workflows/security-notifications-test-workflow.yml#L55-L69) pulls the staging image and replaces (or adds if it's not present) the Notification plugin with the changes in the PR it's running for.

---

I think if you restart the workflow, you will see the problem - my local image build failed for this pull request with JarHell

---

Interesting, it seems like that snapshots are old? These warnings (which led to issues) where also fixed in 3.0.0

```
WARNING: Illegal reflective access by org.apache.hc.core5.util.ReflectionUtils$1 (file:/home/runner/.gradle/caches/modules-2/files-2.1/org.apache.httpcomponents.core5/httpcore5/5.1.5/df9da3a1fa2351c4790245400ed28d78a8ddd3fc/httpcore5-5.1.5.jar) to method sun.security.ssl.SSLEngineImpl.getApplicationProtocol()
> Task :notifications:integTest
WARNING: Illegal reflective access by org.apache.hc.core5.util.ReflectionUtils$1 (file:/home/runner/.gradle/caches/modules-2/files-2.1/org.apache.httpcomponents.core5/httpcore5/5.1.5/df9da3a1fa2351c4790245400ed28d78a8ddd3fc/httpcore5-5.1.5.jar) to method sun.security.ssl.SSLEngineImpl.getApplicationProtocol()
```

---

My local build/tests are still passing. I'm going to leave these as they are for now. If we end up seeing issues in the distribution build later on, I'll follow-up then.